lib/IRGen/IRGen.cpp

```diff
@@ -169,6 +178,29 @@ void swift::performLLVMOptimizations(IRGenOptions &Opts, llvm::Module *Module,
     PMBuilder.addExtension(PassManagerBuilder::EP_EnabledOnOptLevel0,
                            addThreadSanitizerPass);
   }
+
+  if (Opts.SanitizeCoverage != SanitizerCoverageKind::None) {
+    PassManagerBuilder::ExtensionFn extensionFn = nullptr;
+    switch (Opts.SanitizeCoverage) {
+    case SanitizerCoverageKind::Function:
+      extensionFn =
+          addSanitizerCoveragePass<SanitizerCoverageOptions::SCK_Function>;
+      break;
+    case SanitizerCoverageKind::BB:
+      extensionFn = addSanitizerCoveragePass<SanitizerCoverageOptions::SCK_BB>;
+      break;
+    case SanitizerCoverageKind::Edge:
+      extensionFn =
+          addSanitizerCoveragePass<SanitizerCoverageOptions::SCK_Edge>;
+      break;
+    default:
```

Comment:
It's fine to mark the `None` case as `llvm_unreachable("checked above")` or similar. The point is to avoid the `default`.